Makefile

```diff
@@ -235,4 +256,17 @@ lint: $(GOLANGCI_LINT)
 	$(GOLANGCI_LINT) run --timeout=5m -v
 
 lint-charts: $(HELM) # Run helm lint tests
-	helm lint manifest_staging/charts/secretsync
\ No newline at end of file
+	helm lint manifest_staging/charts/secretsync
+
+.PHONY: shellcheck
+shellcheck: $(SHELLCHECK)
+	find . \( -name '*.sh' -o -name '*.bash' \) | xargs $(SHELLCHECK)
+
+## --------------------------------------
+## E2E Testing
+## --------------------------------------
+
+# Run the e2e provider tests
+.PHONY: run-e2e-provider-tests
```

Comment:
This will likely fail on e2e CI job on test-infra. This target requires sync controller to be installed along with SPC and SS object. Creat another dependent target say `e2e-setup` to ensure all the pre-requisite is done before running e2e test. See if you can refactor `local-setup` to reuse some of the installation commands